MRN-Code/coinstac
# Basic File Collections Page

* **Problem:**
   1. The redesign requires a new file collections (aka "My Files", "My Projects") page.
* **Solution:**
   1. Add new IndexedDB wrapper to handle local data - [Dexie](http://dexie.org/)
   1. Add collections list page featuring all local file collections.
   1. Add a tabbed `collection` page with tabs for "About", "Files" and "Consortia".
   1. Functionality for adding multiple local CSV meta files saved to IndexedDB as part of a single collection.
   1. Functionality to map local collection file columns to a member consortia's active pipeline computations.
   1. Automatic input mapping if file contains column name sharing preset input name. 
* **Tests:**
   1. Follow the steps outlined in `packages/coinstac-api-server/README.md` to setup your development environment with RethinkDB, start the GraphQL API, and insert a couple computations into the new database.
   1. Duplicate `packages/coinstac-graphql/config/api-config-sample.json` as `api-config.json`.
   1. Uncomment the contents and fill in a random string for the `jwtSecret`.
   1. Duplicate `packages/coinstac-graphql/config/comp-reg-config-sample.json` as `comp-reg-config.json` and uncomment the contents.
   1. Start COINSTAC UI dev environment as usual. 
   1. Start the application and login with user: `test` & pass: `password`.
   1. Visit the *Collections* navigation item to see a test collection listed.
      1. In the `componentDidMount()` method of `dashboard-home.jsx`, there is a `initTestData()` that pre-populates this test collection. It will overwrite your Dexie contents each time you restart the app/visit the homepage. After adding mappings, comment this line out to ensure those mappings don't get overwritten.
   1. View details for listed collection and navigate through tabs.
   1. Click on the `Add Files` button in the *Files* tab to add a local CSV metafile.
      1. Test meta files can be found in `packages/coinstac-ui/test/e2e/test_files/site1`
      1. An additional option is presented here to deduce the file structure from the directory layout. This functions the same as the metafile option for the time being.
   1. Click the*Consortia* tab to see consortia associated with this Collection (there shouldn't be any to start with).
   1. Choose a consortium from the dropdown beneath the _Add to Consortia_ heading.
   1. Select a source file for the listed covariates and notice that the column is automatically mapped.
